### PR TITLE
[CFM-477] notify group admins that status has changed to republished

### PR DESCRIPTION
**Pull request by anvmn**
## This PR is dependent on #1059 

https://issuetracker.amplexor.com/jira/browse/CFM-477

### CODE REVIEW

**Comment by anvmn** [comment]
selection_111

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc [hunk 1]
```diff
@@ -68,9 +68,33 @@ function c4m_user_notifications_group_status_set_published($node) {
...
+ */
+function _c4m_user_notifications_to_admin_group_status_set_republished($node, $admin_id, $group_former_status) {
+  $message = message_create('c4m_group_republished', array('uid' => $admin_id));
+  c4m_message_load_tokens__c4m_group_republished($message);
+  $message->arguments['@group_former_status'] = $group_former_status;
+
+  $wrapper = entity_metadata_wrapper('message', $message);
+  $wrapper->field_group_node->set($node->nid);
+  // Message will be saved in message_notify_send_message().
+  message_notify_send_message($wrapper->value());
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc]: `message_subscribe_send_message()`

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 2]
```diff
@@ -78,6 +79,20 @@ function _c4m_message_format_welcome_to_group_from_field(&$message, $message_ent
 /**
  * Helper function.
  *
+ * Formats 'c4m_group_republished' message From field, per template requirement.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: @anvmn 
Can you please write the ID of the template ?
thanks

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 3]
```diff
@@ -233,6 +248,44 @@ function c4m_message_load_tokens__c4m_welcome_to_group(Message $message) {
 }
 
 /**
+ * Add callbacks to c4m_group_republished message arguments.
```
Comment (anvmn) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: Add docs

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc [hunk 4]
```diff
@@ -68,9 +68,35 @@ function c4m_user_notifications_group_status_set_published($node) {
...
+function _c4m_user_notifications_to_admin_group_status_set_republished($node, $admin_id, $group_former_status) {
+  global $user;
+  $message = message_create('c4m_group_republished', array('uid' => $admin_id));
+  c4m_message_load_tokens__c4m_group_republished($message);
+  $message->arguments['@group_former_status'] = $group_former_status;
+
+  $wrapper = entity_metadata_wrapper('message', $message);
+  $wrapper->field_group_node->set($node->nid);
+  $wrapper->field_executing_user->set($user->uid);
+  // Message will be saved in message_notify_send_message().
```
Comment (anvmn) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc]: Add space above

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc [hunk 5]
```diff
@@ -68,9 +68,35 @@ function c4m_user_notifications_group_status_set_published($node) {
...
+/**
+ * Helper function.
+ *
+ * Email notification to organic group admin / owner, about status change
+ * from 'archived' to 'published'.
+ */
+function _c4m_user_notifications_to_admin_group_status_set_republished($node, $admin_id, $group_former_status) {
+  global $user;
+  $message = message_create('c4m_group_republished', array('uid' => $admin_id));
+  c4m_message_load_tokens__c4m_group_republished($message);
```
Comment (anvmn) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc]: Add comment

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc [hunk 6]
```diff
@@ -68,9 +68,35 @@ function c4m_user_notifications_group_status_set_published($node) {
...
+  $message->arguments['@group_former_status'] = $group_former_status;
```
Comment (anvmn) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc]: Add comment

**Comment by anvmn** [comment]
## Replaced by #1189

**Pull request closed without merging**